wagtail/images/models.py

```diff
@@ -620,6 +625,9 @@ def find_existing_renditions(
             ]
             for rendition in Rendition.cache_backend.get_many(cache_keys).values():
                 filter = filters_by_spec[rendition.filter_spec]
+                # The retrieved rendition needs to be associated with the current image instance, so that any
+                # locally-set properties such as contextual_alt_text are respected
+                rendition.image = self
```

Comment:
Might be a silly question but this is only done in this specific scenario because… in other scenarios it’s already done elsewhere?

Reply:
Oh, no. It's not done anywhere else